doc/source/tutorials/basictutorial.rst

```diff
@@ -0,0 +1,656 @@
+For instance, printing ``CurrentTime`` would give 60. Not only at ``time:seconds`` = 60,
+also at any other time like ``time:seconds`` = 4000. Eventhough ``time:seconds`` is 4000,
+``CurrentTime`` is still 60.
+
+The ``lock`` command updates constantly the variable, for example: ::
+
+  lock TimeSecondsPlusTen to time:seconds + 10.
+
+If you print ``TimeSecondsPlusTen`` at 60 seconds it will show 70, if you print
+``TimeSecondsPlusTen`` at 4000 seconds it will show 4010.
+
+Using until, lock and wait in an example
+_________________________________________
+
+If we now combine all the command we can make the following piece of code: ::
+
+  set Adder to 0.
+  lock Multiplier to Adder * 2.
+  set TimePlusFive to time:seconds + 5.
+
+  until time:seconds > TimePlusFive {
+    print Multiplier.
+    set Adder to Adder + 1.
+    wait 1.
+  }
+
+So an easy way to read the until loop is to cover what ever is inside of the curly brackets
+and say: until ``time:seconds`` is bigger than our current time plus 5 seconds, print
+``Multiplier``, increase the value of ``Adder`` and wait 1 second.
+
+The outcome of this piece of code is: ::
+
+  0
+  2
+  4
+  6
+  8
+  10
+
+Wait until
+__________
+
+You can also use the ``wait until`` command, this blocks all other code until the
+conditions have been met. ::
+
+  set TimePlusFive to time:seconds + 5.
+  wait until time:seconds > TimePlusFive.
+  print "done waiting".
+
+It will take 5 seconds for ``done waiting`` to show up.
+
+==================
+Lists and lexicons
+==================
+
+::
+
+  set Value1 to 0.
+  set Value2 to 5.
+  set Value3 to 10.
+  set Value4 to 15.
+  set Value5 to 20.
+
+Let's say we want to put these values in a list we want to edit later we can put them into a list by typing the following: ::
+
+  set ValueList to list(Value1, Value2, Value3, Value4, Value5).
+  print ValueList.
+
+This will show: ::
+
+  [0] = 0
+  [1] = 5
+  [2] = 10
+  [3] = 15
+  [4] = 20
+
+As you can see the list goes from 0-4 instead of 1-5. So if you'd want to access ``Value3`` you'd need to look for ``[2]``.
+This can be done as follows: ::
+
+  print ValueList[2]. // shows 10
+
+But let's say you want to print every item in the list you could do: ::
+
+  print ValueList[0]. // shows 0
+  print ValueList[1]. // shows 5
+  print ValueList[2]. // shows 10
+  print ValueList[3]. // shows 15
+  print ValueList[4]. // shows 20
+
+But the problem with this is that you have to know how big the list is and it'd take up a lot space when dealing with big lists. ::
+
+  for Whatever in ValueList {
+    print Whatever.
+  }
+
+  for Value in ValueList {
+    print Value.
+  }
+
+Both pieces of code do **EXACTLY** the same.
+This checks each item in a given list (now called ``Whatever``) and does what the curly brackets contains.
+(For each item in the list called ``ValueList``, which we call ``Whatever``, do whatever is inside of the brackets).
+
+In this case it prints: ::
+
+  0
+  5
+  10
+  15
+  20
+
```

Comment:
Also give an example where the index is itself a variable, maybe?
show that:
```
set x to 3.
print ValueLilst[x].
```
does the same thing as:
```
print ValueList[3].
```